internal/ansible/runner/internal/inputdir/inputdir.go

```diff
@@ -131,6 +132,18 @@ func (i *InputDir) Write() error {
 		return err
 	}
 
+	if strings.HasPrefix(i.CmdLine, string("'")) && i.CmdLine[0] == i.CmdLine[len(i.CmdLine)-1] {
+		i.CmdLine = i.CmdLine[1 : len(i.CmdLine)-1]
+	}
+
```

Comment:
I did not get what it is doing? could you please clarifies? Maybe a comment in the code.

Reply:
The arbitrary args input may come as '<<Args command>>', but we do not want to parse it like that. So I checked if the command has ' ' wrapped around it and removed it, so it can execute the command correctly. I will put this in the comments.

Reply:
Note that Go has a lib called `strings` which provide funcs for we manipulated the data. 
We should try to use the recourses of the language always since they usually are customized and more understandable.  It is very important to write a code that any person can read and check what is going without too much effort. 

In this way, here we can do such the followjng example : 

```go
      package main

import (
	"fmt"
	"strings"
)

func main() {

	argValue := "'value'"
	singleQuotes := "'"

	argValue = strings.TrimLeft(argValue, singleQuotes)
	argValue = strings.TrimRight(argValue, singleQuotes)	
	
	fmt.Println(argValue) // It will print value without the single quote. 
}
```

Could we improve that in order to keep a better maintenance ability?

Reply:
Will do, thanks for the suggestion.

Reply:
@camilamacedo86, just realized a flaw with the above solution. Let's consider a use case:

```
argValue := "'--ansible-args='--vault-password-file /opt/ansible/pwd.yml''"
	
argValue = strings.TrimLeft(argValue,"'")
argValue = strings.TrimRight(argValue, "'")
	
fmt.Println(argValue)
```

This will output: 
`--ansible-args='--vault-password-file /opt/ansible/pwd.yml`

But we want
`--ansible-args='--vault-password-file /opt/ansible/pwd.yml'`

Since it is possible that the last TWO characters may be`'`, but we only want to remove the last one. Trimming from the left will remove all such characters, leaving the quotations open and it will not be able to parse the argument correctly.

What are your thoughts?

Reply:
So, all singleQuotes should be removed. Am I right? If yes, we can just ` strings.ReplaceAll(value, "'", "") which shows more accurate indeed.

Reply:
No, we are trying to remove only the first and last single quotes if the string is wrapped with them. Only if the first and last characters are single quotes, then we want to remove those. Rest all single quotes are used by the argument and hence should not be removed.

Reply:
Hi @VenkatRamaraju, 

I saw that you added a comment so it will help already. I still think that would be possible to do the same by using HasPrefix, HasSufix and the replaces. but it is fine .. I am ok with 👍 we always can improve in follow-ups as well